crates/fuel-core/src/graphql_api/da_compression.rs

```diff
@@ -65,14 +72,23 @@ where
     Ok(())
 }
 
-struct CompressTx<'a, Tx> {
-    db_tx: &'a mut Tx,
+pub struct DbTx<'a, Tx> {
+    pub db_tx: &'a mut Tx,
+}
```

Comment:
What's the benefit we get from this wrapper? Couldn't we just use `&'a mut Tx` directly in `DecompressDbTx` in the same way it used to be in `CompressTx`?

Reply:
`DbTx` is used to share `TemporalRegistry` implementation between `CompressDbTx` and `DecompressDbTx`. If we'd be using `&'a mut Tx` directly, then the `TemporalRegistry` implentation would have to be duplicated for both of them.

Reply:
But couldn't we implement `TemporalRegistry` directly on `&'a mut Tx` instead?

Reply:
Ah right but yeah that gets more convoluted since `Tx` is generic, so we'd have to make a blanket implementation or battle with a bunch of extra `where` clauses in any `impl` blocks using this. Sure I see why this might be simpler then.